### PR TITLE
experiment: embedding classifier prototype (refs #36)

### PR DESCRIPTION
**Pull request by maciej-makowski**
## Summary

Experimental eval for [#36](https://github.com/maciej-makowski/are-they-hiring/issues/36): can an Ollama embedding + sklearn head beat the current `qwen2.5:1.5b` few-shot LLM classifier (baseline: **97.8% accuracy / 69.8% precision / 72.5% recall**) on the 1381 hand-labelled titles?

**TL;DR: no embedding variant strictly dominates the baseline.** The LLM classifier sits on the Pareto frontier — every embedding configuration trades precision for recall or vice versa. Full write-up in [`scripts/embed_classify_results.md`](scripts/embed_classify_results.md).

### Results (5-fold CV, L2-normalized)

| Combination | Acc | Prec | Rec |
| --- | ---: | ---: | ---: |
| **qwen2.5:1.5b few-shot (baseline)** | **97.8%** | **69.8%** | **72.5%** |
| all-minilm + LinearSVC no-balance     | 97.9% | 89.3% | 49.0% |
| all-minilm + LinearSVC balanced       | 97.4% | 60.0% | 88.2% |
| all-minilm + LR thr=0.20 (no-balance) | 97.5% | 73.5% | 49.0% |
| mxbai-embed-large + LinearSVC no-bal  | 97.4% | 82.6% | 37.3% |
| nomic-embed-text (any variant)        | collapses or predicts all-zero |

- **Best precision:** `all-minilm + LinearSVC (no class_weight)` — 89.3% precision, but only 49% recall.
- **Best recall:** `all-minilm + LinearSVC (balanced)` — 88.2% recall, but 60% precision.
- **Nearest-to-baseline:** `all-minilm + LR thr=0.20` — 73.5% precision, 49% recall. Matches precision, loses 23 pp of recall.

### Pi viability

- `all-minilm` is 46 MB (vs 940 MB for qwen2.5:1.5b) and embeds ~190 titles/s on GPU. Classifier pickle is ~15 KB. Free real-estate on the Pi.
- `mxbai-embed-large` is 670 MB, strictly worse on this dataset than `all-minilm`.
- `nomic-embed-text` is unusable here, prefix or no prefix.

### Recommendation: don't ship as a straight swap

The baseline is Pareto-optimal across the eval points we tested. The interesting follow-up — which this prototype doesn't implement — is a **two-stage pipeline**: run the cheap balanced-SVC head as a pre-filter, then send the ~3% it flags `yes` to the LLM for a precision check. That would cut ~98% of negative-class LLM calls while retaining the baseline's precision. Worth opening a follow-up issue if we want to pursue it.

### What changed

- `scripts/embed_classify_experiment.py` — new (embeds titles, trains LR + SVC heads with/without class_weight, does 80/20 + 5-fold CV + threshold sweep, caches both embeddings and fitted classifiers).
- `scripts/embed_classify_results.md` — new (full results table + recommendation).
- `.gitignore` — ignore regenerable `scripts/cache/`.
- `pyproject.toml` / `uv.lock` — add `scikit-learn` and `numpy` as dev deps.

No production code (`src/`, `Containerfile.ollama`) touched.

## Test plan

- [x] `uv run ruff check src/ tests/ scripts/` — clean
- [x] `uv run ruff format --check src/ tests/ scripts/` — clean
- [x] `make test` — 77 integration tests pass (unchanged, nothing to regress)
- [x] Pre-commit hooks pass on the commit (`ruff check`, `ruff format`, integration tests)
- [x] Script runs end-to-end against pop-os Ollama; cache round-trips correctly (second run hits cache, no re-embedding)

### How to reproduce

Ground truth CSV needs to be at `/tmp/roles-to-label.csv` (1381 rows, `title` + `your_label` columns). Then:

```
uv run python scripts/embed_classify_experiment.py --l2-normalize --cv
```

Default Ollama host is `http://pop-os.home.lan:11434`. Override with `--ollama-host http://localhost:11435` if needed. The script will pull any missing embedding models automatically.

_Not intended for merge — if we act on the recommendation, this would motivate a separate integration PR (or a close for #36)._

### CODE REVIEW

**Comment by maciej-makowski** [comment]
Closing without merging — this was a research-only experiment (see issue #36).

Results: on the 1381-title ground truth, `all-minilm + LinearSVC (no class_weight)` hit 89.3% precision / 49% recall. Beats qwen2.5:1.5b on precision (+19.5 pp) but loses 23.5 pp of recall. qwen2.5:1.5b Pareto-dominates across the operating points we measured.

The actionable finding — `all-minilm + LinearSVC balanced` as a pre-filter before the LLM — is tracked in #45.
